avocado/core/job.py

```diff
@@ -597,11 +613,11 @@ def run_tests(self):
         self.args.standalone = True
         self.args.show = ["test"]
         output.reconfigure(self.args)
-        self.job = Job(self.args)
-        exit_status = self.job.run()
-        if self.args.remove_test_results is True:
-            shutil.rmtree(self.job.logdir)
-        sys.exit(exit_status)
+        with Job(self.args) as self.job:
+            exit_status = self.job.run()
+            if self.args.remove_test_results is True:
+                shutil.rmtree(self.job.logdir)
+            sys.exit(exit_status)
```

Comment:
Suggestion: move the `sys.exit(exit_status)` to outside the job context handler.  This changes the order in which things are called, and `cleanup()` gets called before `sys.exit()`, which makes more sense IMO.

Reply:
sure